vars/runATH.groovy

```diff
@@ -198,10 +198,33 @@ private void test(discriminator, commandBase, localSnapshots, localPluginsStashN
     unstashResources(localSnapshots, localPluginsStashName)
     athContainerImage.inside(containerArgs) {
         realtimeJUnit(testResults: 'target/surefire-reports/TEST-*.xml', testDataPublishers: [[$class: 'AttachmentPublisher']]) {
-            def command = './set-java.sh $java_version && eval "$(./vnc.sh)" && export DISPLAY=$BROWSER_DISPLAY && export SHARED_DOCKER_SERVICE=true && export EXERCISEDPLUGINREPORTER=textfile && ' + prepareCommand(commandBase, discriminator, localSnapshots, localPluginsStashName)
+            /*
+            If you want to compile with java > 8 and have all needed, do it
+            If you want to compile with java > 8 and you lack the set-java script, FAILURE
+            If other case (java8 and lack set-java), write a message in the log
+             */
+            def command = '''
+            if [ -x ./set-java.sh ] && [ -n "$java_version" ] && [ "$java_version" -gt "8" ]; then
+                ./set-java.sh $java_version;
+            elif [ -n "$java_version" ] && [ "$java_version" -gt "8" ] && [ ! -x ./set-java.sh ]; then
+                echo "ERROR: ./set-java.sh not found because you are using old ATH sources and \\$java_version = $java_version specified. We cannot run on this JDK";
+                exit 1;
+            elif [ ! -x ./set-java.sh ]; then
+                echo "INFO: ./set-java.sh not found because you are using old ATH sources, please consider to update ATH sources and docker image";
```

Comment:
Due to the few valuable outcome and the time required to add a meanful GitHub status on this case, I opted to leave it as a message. Java8 with old ATH sources is not failing actually.